src/Promitor.Core.Scraping/Configuration/Model/Metrics/AzureResourceDefinition.cs

```diff
@@ -0,0 +1,30 @@
+﻿namespace Promitor.Core.Scraping.Configuration.Model.Metrics
+{
+    /// <summary>
+    /// Describes a resource in Azure that can be scraped. Inheriting classes can add whatever
+    /// additional information is required to scrape a particular Azure resource.
+    /// </summary>
+    public abstract class AzureResourceDefinition
+    {
+        protected AzureResourceDefinition(ResourceType resourceType)
+        {
+        }
+
+        protected AzureResourceDefinition(ResourceType resourceType, string resourceGroupName)
+        {
+            ResourceType = resourceType;
+            ResourceGroupName = resourceGroupName;
+        }
+
+        /// <summary>
+        ///     Type of resource that is configured
+        /// </summary>
+        public ResourceType ResourceType { get; }
+
+        /// <summary>
+        ///     Specify a resource group to scrape that defers from the default resource group.
+        ///     This enables you to do multi-resource group scraping with one configuration file.
+        /// </summary>
+        public string ResourceGroupName { get; set; }
```

Comment:
Unfortunately we do because of https://github.com/tomkerkhove/promitor/blob/master/src/Promitor.Core.Scraping/Configuration/Providers/MetricsDeclarationProvider.cs#L37.

If we wanted to remove the setter I guess we'd need to rethink how the defaulting worked.

Reply:
Just ignore that comment - as you pointed out the defaulting code in MetricsDeclarationProvider isn't actually required, so I was able to make this private.

Reply:
Awesome, thanks!